http: Task_CRUD fails race tests intermittently

```
--- FAIL: TestTaskService (2.31s)
    --- FAIL: TestTaskService/TaskService (0.00s)
        --- PASS: TestTaskService/TaskService/Task_Meta_Updates (0.46s)
        --- FAIL: TestTaskService/TaskService/Task_CRUD (0.46s)
            servicetest.go:182: task not found

```

--- a/http/task_service.go
+++ b/http/task_service.go
@@ -271,47 +271,14 @@ func (h *TaskHandler) handleGetTasks(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var tasks []*platform.Task
-
-	if req.filter.User != nil {
-		ownedTasks, _, err := h.UserResourceMappingService.FindUserResourceMappings(
-			ctx,
-			platform.UserResourceMappingFilter{
-				UserID:       *req.filter.User,
-				UserType:     platform.Owner,
-				ResourceType: platform.TasksResourceType,
-			},
-		)
-		if err != nil {
-			err = &platform.Error{
-				Err: err,
-				Msg: "failed to pull user's resources",
-			}
-			EncodeError(ctx, err, w)
-			return
+	tasks, _, err := h.TaskService.FindTasks(ctx, req.filter)
+	if err != nil {
+		err = &platform.Error{
+			Err: err,
+			Msg: "failed to find tasks",
 		}
-		for _, ownedTask := range ownedTasks {
-			task, err := h.TaskService.FindTaskByID(ctx, ownedTask.ResourceID)
-			if err != nil {
-				err = &platform.Error{
-					Err: err,
-					Msg: "failed to find tasks",
-				}
-				EncodeError(ctx, err, w)
-				return
-			}
-			tasks = append(tasks, task)
-		}
-	} else {
-		tasks, _, err = h.TaskService.FindTasks(ctx, req.filter)
-		if err != nil {
-			err = &platform.Error{
-				Err: err,
-				Msg: "failed to find tasks",
-			}
-			EncodeError(ctx, err, w)
-			return
-		}
+		EncodeError(ctx, err, w)
+		return
 	}
 
 	if err := encodeResponse(ctx, w, http.StatusOK, newTasksResponse(ctx, tasks, h.LabelService)); err != nil {

--- a/task/platform_adapter.go
+++ b/task/platform_adapter.go
@@ -57,9 +57,6 @@ func (p pAdapter) FindTasks(ctx context.Context, filter platform.TaskFilter) ([]
 		org.ID = *filter.OrganizationID
 	}
 	if filter.Organization != "" {
-		if filter.Organization == "non-existant-org" {
-			fmt.Println("here??")
-		}
 		err := p.populateOrg(ctx, &org)
 		if err != nil {
 			return nil, 0, err
@@ -79,10 +76,15 @@ func (p pAdapter) FindTasks(ctx context.Context, filter platform.TaskFilter) ([]
 			return nil, 0, err
 		}
 
-		var tasks []*platform.Task
+		tasks := make([]*platform.Task, 0, len(ownedTasks))
 		for _, ownedTask := range ownedTasks {
 			storeTask, meta, err := p.s.FindTaskByIDWithMeta(ctx, ownedTask.ResourceID)
 			if err != nil {
+				// It's possible we had an entry in the list a moment ago and it's since been deleted.
+				if err == backend.ErrTaskNotFound {
+					// If so, just move on.
+					continue
+				}
 				return nil, 0, err
 			}
 			task, err := p.toPlatformTask(ctx, *storeTask, meta)
@@ -91,8 +93,8 @@ func (p pAdapter) FindTasks(ctx context.Context, filter platform.TaskFilter) ([]
 			}
 
 			tasks = append(tasks, task)
-
 		}
+
 		return tasks, len(tasks), nil
 	}
 
@@ -427,7 +429,6 @@ func (p *pAdapter) populateOrg(ctx context.Context, org *platform.Organization) 
 		org.Name = o.Name
 	} else {
 		o, err := p.orgSvc.FindOrganization(ctx, platform.OrganizationFilter{Name: &org.Name})
-		fmt.Println(o, err, "BLARG")
 		if err != nil {
 			return err
 		}

